src/main/kotlin/com/exactpro/th2/codec/oracle/logminer/LogMinerTransformer.kt

```diff
@@ -164,31 +586,146 @@ class LogMinerTransformer(private val config: LogMinerConfiguration) : IPipeline
             setProtocol(this@toBuilderWithoutBody.protocol)
         }
 
-        internal fun Function.toReadable(onWarning: (String) -> Unit): Map<String, Any?> {
-            return hashMapOf(
-                "function" to name,
-                "parameters" to parameters.map { it.toReadable(onWarning) }
-            )
+        private fun Throwable.extractMessage(): String? = message?.lines()?.first()
+
+        private inline fun <T> MutableList<T>.removeLastWhile(predicate: (T) -> Boolean): List<T> {
+            if (isEmpty())
+                return emptyList()
+            val iterator = listIterator(size)
+            while (iterator.hasPrevious()) {
+                if (!predicate(iterator.previous())) {
+                    iterator.next()
+                    val expectedSize = size - iterator.nextIndex()
+                    if (expectedSize == 0) return emptyList()
+                    return ArrayList<T>(expectedSize).apply {
+                        while (iterator.hasNext()) {
+                            add(iterator.next())
+                            iterator.remove()
+                        }
+                    }
```

Comment:
Just for information: you could use `buildList` here
```suggestion
                    return buildList(expectedSize) {
                        while (iterator.hasNext()) {
                            add(iterator.next())
                            iterator.remove()
                        }
                    }
```

Reply:
Thank you